docs/source/tech_specs.rst

```diff
@@ -0,0 +1,111 @@
+
+Array index is probably the best option - although they are a little fragile,
+they're no more fragile than path references, and require far less overhead
+than unique keys.
+
+We need to look at this beyond atoms and bonds. Especially in workflows we can reuse pieces of data 
+from previous tasks in the workflow. Instead of repeating we can use referencing.
+
+See also: http://stackoverflow.com/q/4001474/1958900
+
+How do we uniquely specify physical units?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+For instance, velocity might be "angstrom/fs" Alternatives:
+
+ 1) Require units in the form {:code:`unit_name:exponent`}, e.g. :code:`atom.velocity.units={'angstrom':1, 'fs':-1}`
+ 2) Allow strings of the form :code:`atom.velocity.units="angstrom/fs"`, but require that units be chosen from a specific list of specifications
+ 3) Allow strings of the form :code:`atom.velocity.units="angstrom/fs"`, and require file parsers to parse the units according to a specified syntax
+ 
+ BDJ Note: There are multiple standards specifications for units, and conversions. If done right in a schema, you can use JSON-LD to
+ link to the actual standards definition. Some examples of what I have done, which aligns with CML:
+ 
+.. code:: python
+
+    "orbitalEnergy": {"units": "Hartree", "value": 0.935524}
+    "shieldingAnisotropy": {"units": "ppm","value": 17.5292}
+
+
+JSON and HDF5
+~~~~~~~~~~~~~
+
+The object specifications in this document are tailored to JSON, but can be
+easily stored in an HDF5 file as well. HDF5 is, like JSON, hierarchical and
+self-describing. These similarities make it easy to perform 1-to-1
+transformations between well-formed JSON and a corresponding HDF5
+representation.
+
+Unlike JSON, HDF5 is binary and requires custom libraries to read, but has far
+better performance and storage characteristics for numerical data. We will
+provide tools to easily interconvert files between JSON and HDF5. Applications
+that support this format should always provide JSON support; ones that require
+high performance should also support the HDF5 variant.
+
+Closed Questions
+----------------
+
+Store large collections of objects
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+How do we store large lists of objects (such as lists of atoms or bonds?)
+
+ 1) As a table of values (these values do very well in HDF5 as well)
+ 2) As a set of arrays
+ 3) As a list of objects (works well in HDF5 too, would be worth brining in a HDF5 expert)
+
+Examples:
+
+.. code:: python
+
+    // 1) Storing fields as tables: creates an mmCIF/PDB-like layout
+    {atoms={type:'table[atom]',
+            fields=['name', 'atomic_number', 'mass/Dalton', 'residue_index', 'position/angstrom', 'momentum/angstrom*amu*fs^-1']
+            entries=[
+              ['CA', 6, 12.0, 0, [0.214,12.124,1.12], [0,0,0]],
+              ['N', 7, 14.20, 0, [0.214,12.124,1.12], [0,0,0]],
+              ...}
+    
+    // 2) Storing fields as arrays: much more compact, but harder to read and edit
```

Comment:
Perhaps its harder to read/edit as a human, but not in a program. Please don't use this language. As @loriab mentioned, this was the choice at the workshop and there are a lot of benefits on the "writing code" side.

Reply:
I was mostly porting over current Markdown documentation to RST as an example, but there seems to be bringing up a bit of old discussion that is indeed solved. Should we covert the solved topics over to a simpler discussion of why the choice was made?

Reply:
Yes, I think that's a good idea. As people implement, I don't think there should be ambiguity and I think a discussion of the choice is good.

(Incidentally, I'd suggest when the schema is done, that this be published in J. Cheminf.)